### PR TITLE
Aliases Part 1

### PR DESCRIPTION
**Pull request by louietyj**
**Done**
- Alias view (`alias`)
- Set aliases (`alias fromthis tothis`)
- Disambiguation

---

**Not done** (separate PR):
- Unalias
- Actually make use of the alias in InputHandler

@irvinlim Do you think aliases should be stored in the DB or configSettings? Honestly I don't really know why they're two different files in the first place =x

### CODE REVIEW

**Comment by irvinlim** [comment]
@louietyj Since we have to allow the user to specify a custom storage location, the config file will save the storage path as its config. The config will be in the same location as the jar. This one was original behaviour one, didn't change.

I think the aliases should be in Config? Sounds better to me, because program settings should be independent of saved data?

**Comment by irvinlim** [comment]
@louietyj Are you planning to validate aliases to a set of allowed commands? Like, so that I can't `alias lksjdlf alskjdas`

File: src/main/java/seedu/todo/controllers/AliasController.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+import java.util.List;
+import java.util.Map;
+
+import seedu.todo.MainApp;
+import seedu.todo.commons.core.Config;
+import seedu.todo.commons.exceptions.CannotConfigureException;
+import seedu.todo.commons.util.ConfigUtil;
+import seedu.todo.controllers.concerns.Renderer;
+import seedu.todo.models.TodoListDB;
+import seedu.todo.ui.UiManager;
```
Comment (irvinlim) [inline src/main/java/seedu/todo/controllers/AliasController.java]: Should remove all these unused imports

**Comment by ChaseTiong** [comment]
I think we have to validate it.

**Comment by louietyj** [comment]
Not a planned feature because of the complexity of having to keep a list of whitelisted commands. I can't really think of an elegant way to keep an authoritative list of what can and cannot be aliased.

If we really want, we can exhaustively go add everything into a list I guess.

File: src/main/java/seedu/todo/controllers/AliasController.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+ * Controller to declare aliases
+ * 
+ * @author louietyj
+ *
+ */
+public class AliasController implements Controller {
+    
+    private static final String NAME = "Alias";
+    private static final String DESCRIPTION = "Shows current aliases or updates them.";
+    private static final String COMMAND_SYNTAX = "alias [<setting> <value>]";
```
Comment (irvinlim) [inline src/main/java/seedu/todo/controllers/AliasController.java]: Syntax should be `alias [<command> <value>]` or something

**Comment by louietyj** [comment]
Note that we may want to alias tokens too, which makes our life much harder. And I haven't really figured out what to do with cyclic aliases yet, i.e. abc -> def -> ghi -> abc.

**Review by irvinlim — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/todo/controllers/AliasController.java (hunk 2 above)
Comment (louietyj) [reply src/main/java/seedu/todo/controllers/AliasController.java]: Oops copy-pasta

**Comment by louietyj** [comment]
> I think the aliases should be in Config? Sounds better to me, because program settings should be independent of saved data?

So we draw the line at TodoListDB storing raw data meant to be imported/exported and Config storing individual user preferences yeah?

Will move aliases into Config in a separate PR. This PR is getting big.

**Review by irvinlim — APPROVED** [review]

**Pull request merged**